nautobot_device_lifecycle_mgmt/filters.py

```diff
@@ -141,6 +139,121 @@ def search(self, queryset, name, value):  # pylint: disable=unused-argument, no-
+        queryset=SoftwareLCM.objects.all(),
+        to_field_name="version",
+        label="Software (version)",
+    )
+    device_types_id = django_filters.ModelMultipleChoiceFilter(
+        field_name="device_types",
+        queryset=DeviceType.objects.all(),
+        label="Device Types",
+    )
+    device_types = django_filters.ModelMultipleChoiceFilter(
+        field_name="device_types__model",
+        queryset=DeviceType.objects.all(),
+        to_field_name="model",
+        label="Device Types (model)",
+    )
+    inventory_items_id = django_filters.ModelMultipleChoiceFilter(
+        field_name="inventory_items",
+        queryset=InventoryItem.objects.all(),
+        label="Inventory Items",
+    )
+    inventory_items = django_filters.ModelMultipleChoiceFilter(
+        field_name="inventory_items__name",
+        queryset=InventoryItem.objects.all(),
+        to_field_name="name",
+        label="Inventory Items (name)",
+    )
+    object_tags_id = django_filters.ModelMultipleChoiceFilter(
+        field_name="object_tags",
+        queryset=Tag.objects.all(),
+        label="Object Tags",
+    )
+    object_tags = django_filters.ModelMultipleChoiceFilter(
+        field_name="object_tags__slug",
+        queryset=Tag.objects.all(),
+        to_field_name="slug",
+        label="Object Tags (slug)",
+    )
+    device_name = django_filters.CharFilter(method="device", label="Device Name")
+    device_id = django_filters.CharFilter(method="device", label="Device ID")
+    inventory_item_id = django_filters.CharFilter(method="inventory_item", label="InventoryItem ID")
+
+    class Meta:
+        """Meta attributes for filter."""
+
+        model = SoftwareImageLCM
+
+        fields = [
+            "image_file_name",
+            "software",
+            "software_version",
+            "image_file_checksum",
+            "download_url",
+            "device_types",
+            "inventory_items",
+            "object_tags",
+            "default_image",
+            "device_name",
+            "inventory_item_id",
+        ]
+
+    def search(self, queryset, name, value):  # pylint: disable=unused-argument, no-self-use
+        """Perform the filtered search."""
+        if not value.strip():
+            return queryset
+
+        qs_filter = Q(image_file_name__icontains=value) | Q(software__version__icontains=value)
+        return queryset.filter(qs_filter)
+
+    def device(self, queryset, name, value):  # pylint: disable=no-self-use
+        """Search for software image for a given device."""
+        value = value.strip()
+        if not value:
+            return queryset
+
+        if name == "device_name":
+            devices = Device.objects.filter(name=value)
+        elif name == "device_id":
+            devices = Device.objects.filter(id=value)
+        else:
+            devices = Device.objects.none()
+
+        if devices.count() != 1:
+            return queryset.none()
```

Comment:
what is the difference between returning `queryset` (line 225) and `queryset.none()`?

Reply:
It returns empty `queryset` if more than one device is found.
For instance you have devices:
`device-1a`
`device-1b`
your query is `device-1`, but there is no such device.